src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -30874,6 +30874,84 @@ public class Derived : Base
             comp.VerifyDiagnostics();
         }
 
+        [Fact, WorkItem(42169, "https://github.com/dotnet/roslyn/issues/42169")]
+        public void NotNullWhenTrue_WithMaybeNull_Overriding()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+public class C
+{
+    internal virtual bool TryGetValueCore<TKey, TValue>(TKey key, [MaybeNull] [NotNullWhen(true)] out TValue value)
+        where TKey : class
+    {
+        throw null!;
+    }
+}
+public class Derived : C
+{
+    internal override bool TryGetValueCore<TKey, TValue>(TKey key, [MaybeNull] [NotNullWhen(true)] out TValue value)
+    {
+        throw null!;
+    }
+}
+";
+            var comp = CreateNullableCompilation(new[] { MaybeNullAttributeDefinition, NotNullWhenAttributeDefinition, source });
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(42169, "https://github.com/dotnet/roslyn/issues/42169")]
+        public void NotNullWhenFalse_WithMaybeNull_Overriding()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+public class C
+{
+    internal virtual bool TryGetValueCore<TKey, TValue>(TKey key, [MaybeNull] [NotNullWhen(false)] out TValue value)
+        where TKey : class
```

Comment:
is TKey necessary to reproduce the issue (before the compiler change)?

Reply:
It's not. This came from the use case I ran into in an adoption PR. I'll remove the parameter :-)